src/Groups/GAPGroups.jl

```diff
@@ -63,10 +63,7 @@ export
 # the custom group types and isos they define should be added to the arrays
 # _gap_group_types resp. _iso_function
 
-
-function group_element(G::T, x::GapObj) where T <: GAPGroup
-  return GAPGroupElem{T}(G, x)
-end
+group_element(G::T, x::GapObj) where T <: GAPGroup = BasicGAPGroupElem{T}(G,x)
```

Comment:
Perhaps you can extract the change(s) which introduce `BasicGAPGroupElem` into a standalone PR, which then would be fairly small and could be merged quickly; that way, this PR here would not touch quite as many files.